File: Arcade/Intro/3 - Smooth Sailing/9 - All Longest Strings/allLongestStrings.java
String[] allLongestStrings(String[] inputArray) {

    int maxSequence = 0;
    ArrayList<String> preResult = new ArrayList<>();
    
    //goes through the array
    for(int i = 0; i < inputArray.length; i++)
    {
        //if a bigger length of string is found
        if(maxSequence < inputArray[i].length())
        {
            //new arrayList and update the size
            preResult = new ArrayList<>();
            maxSequence = inputArray[i].length();
        }
        if(maxSequence == inputArray[i].length())
        {
            //add the string
            preResult.add(inputArray[i]);
        }
    }
    
    //arrayList to array
    String[] result = new String[preResult.size()];
    result = preResult.toArray(result);
    
    return result;
}

File: Arcade/Intro/3 - Smooth Sailing/9 - All Longest Strings/allLongestStrings.cpp
std::vector<std::string> allLongestStrings(std::vector<std::string> inputArray) {
    
    int maxSequence = 0;
    std::vector<std::string> result;
    
    for(int i = 0; i < inputArray.size(); i++)
    {
        //a string with a bigger length is found
        if(maxSequence < inputArray.at(i).length())
        {
            //reset the array and update the max sequence value
            result.resize(0);
            maxSequence = inputArray.at(i).length();
        }
        if(maxSequence == inputArray.at(i).length())
        {
            //add the string
            result.push_back(inputArray.at(i));
        }
    }
    
    return result;
}
